package/abs/rancher-abs

```diff
@@ -395,24 +393,37 @@ mountdest() {
 
     if [ $? -ne 0 ]; then
         if [ "$(echo ${mount_result} | grep 'is already mounted')" ]; then
-            log "> fail, ${device_path} has already been mounted" /tmp/rancher_abs.log
+            log "> fail, ${device_path} has already been mounted"
             print_error "${mount_result}"
         fi
 
+        local try_fix=0
+        # for DiskCatagory: cloud
         if [ "$(echo ${mount_result} | grep 'wrong fs type')" ]; then
-            log "> fail, try to fix. ${mount_result}" /tmp/rancher_abs.log
+            try_fix=1
+        # for DiskCatagory: cloud_ssd
+        elif [ "$(echo ${mount_result} | grep 'unknown filesystem type')" ]; then
+            try_fix=1
+        fi
+
+        if [ "${try_fix}"x = "1"x ]; then
```

Comment:
Is there a possibility that you would format a disk contained valid data but you simply don't have the driver for it at the moment?

Reply:
It would be possible.
I think there are two problems here,
1.  only some certain types of error are captured here. For example, when 'device does not exist' error occurs the mount function ignores the error and returns a success.
2. We should do the fs formatting on volume creation instead of mounting.